webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsResponderADM.scala

```diff
@@ -838,62 +819,67 @@ final case class ProjectsResponderADM(actorDeps: ActorDeps, cacheServiceSettings
             updatedProject
           )
 
-      _ = if (projectUpdatePayload.shortname.isDefined) {
-            val unescapedShortName: String = stringFormatter.fromSparqlEncodedString(projectUpdatePayload.shortname.get)
+      _ = projectUpdatePayload.shortname.map { shortname =>
+            val unescapedShortName: String = stringFormatter.fromSparqlEncodedString(shortname.value)
             if (updatedProject.shortname != unescapedShortName)
               throw UpdateNotPerformedException(
                 "Project's 'shortname' was not updated. Please report this as a possible bug."
               )
           }
```

Comment:
I support your effort in refactoring the code base into a more functional style.
However, when I read `map` I would generally expect a pure function to be used in its code block.
The `throw ...Exception` in the `map` code block is surprising at best. 

How about changing it to something like:
```scala
projectUpdatePayload.shortname
    .map(_.value)
    .map(stringFormatter.fromSparqlEncodedString)
    .filter(_ != updatedProject.shortname)
   .getOrElse(throw UpdateNotPerformedException("Project's 'shortname' was not updated. Please report this as a possible bug.")
```

Reply:
I really like your suggestion and I will apply it. For the simple values it is easy. What about more complex ones like `descriptions` of type `Seq` where you have a sequence that needs to be processed inside a map again. I came up with the code below (see also https://github.com/dasch-swiss/dsp-api/pull/2394/commits/aa17b20c037d6992cbd98e0abf291978fc7184ad) but it seems very hard to understand (because of the map inside the map). Do you maybe have any suggestion to improve it? I tried to model it inside a for comprehension but it doesn't look better to me.

```scala
_ = projectUpdatePayload.description
    .map(_.value)
    .map(_.map(d => V2.StringLiteralV2(stringFormatter.fromSparqlEncodedString(d.value), d.language)))
    .filter(updatedProject.description.diff(_).nonEmpty)
    .getOrElse(
        throw UpdateNotPerformedException(
            "Project's 'description' was not updated. Please report this as a possible bug."
        )
    )
```